include/lbann/utils/threads/memory.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef LBANN_MEMORY_HPP_
+#define LBANN_MEMORY_HPP_
+
+#include <memory>
+
+namespace lbann {
+
+#if __cplusplus < 201402L
+
+/** \brief Local definition of make_unique for non-C++14 compilers */
+template <typename T, typename... Ts>
+std::unique_ptr<T> make_unique(Ts&&... params)
```

Comment:
Oh hey, my memory file. Since we have `lbann::make_unique`, we should use it. That should be a different PR, though. @timmoon10

Reply:
Nifty. Pinging Issue #155.